Source/WebKitLegacy/mac/WebView/WebView.mm

```diff
@@ -1520,6 +1521,7 @@ - (void)_commonInitializationWithFrameName:(NSString *)frameName groupName:(NSSt
         makeUniqueRef<WebCore::DummyStorageProvider>(),
         makeUniqueRef<WebCore::DummyModelPlayerProvider>(),
         WebCore::EmptyBadgeClient::create(),
+        LegacyHistoryItemClient::create(),
```

Comment:
Do we really need this LegacyHistoryItemClient? Why can't we just use globalHistoryItemClient() consistently in WebKitLegacy?

Reply:
I'll use a singleton.  They're kind of equivalent because it only calls a global function, but you're right a singleton would have fewer allocations with many WebViews.